### PR TITLE
(deps): Dependency Upgrade

### LINKED ISSUE #708: Update Project to the latest version of its dependencies
**I'm submitting a ...**
- [x] bug report
- [X] feature request

**Current behavior:**

Currently project runs on older version of Angular(Angular v4.3) and other dependencies. This is leading to dependency mismatch in travis build.

**Expected behavior:**

It should work on latest Angular v6.

**Screenshots of the issue:**

Not Applicable.

### PR DESCRIPTION
**Pull request by hemantjadon**
**Changes proposed in this pull request**
- Upgrades dependency of the project. 
- Angular to V6
- Rxjs to V6

**Link to live demo:** http://pr-728-fossasia-loklaksearch.surge.sh

**Closes #708**

### CODE REVIEW

**Comment by simsausaurabh** [comment]
@hemantjadon Thank you for the patch for updating project. I am making changes in travis for node version compatibility and yarn updation so that the travis could pass.

File: .travis.yml [hunk 1]
```diff
@@ -27,11 +27,10 @@ matrix:
       - ng build --prod --aot --build-optimizer
 
   - script:
-      - ng test --single-run --code-coverage --reporters=coverage-istanbul
+      - ng test --watch=false --code-coverage
 
   - script:
```
Comment (simsausaurabh) [inline .travis.yml]: Remove script tag and merge its content into above script tag.

File: .travis.yml [hunk 2]
```diff
@@ -27,11 +27,10 @@ matrix:
       - ng build --prod --aot --build-optimizer
 
   - script:
```
Comment (simsausaurabh) [inline .travis.yml]: remove ```-``` and shift the content to left

File: .travis.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ dist: trusty
 language: node_js
 
 node_js:
-  - 6
+  - 8.11
```
Comment (simsausaurabh) [inline .travis.yml]: Remove matrix, include, and services and shift the content to left align and remove ```-``` from left side of pameters(before_script and script).

**Comment by simsausaurabh** [comment]
@hemantjadon Although travis build as passed, but the project is breaking in function.

1. Results are not getting displayed for normal queries
2. Improper alignment
3.  UI bugs

screenshot from 2018-05-29 17-56-35

screenshot from 2018-05-29 17-47-52

screenshot from 2018-05-29 17-48-16

screenshot from 2018-05-29 17-48-37

screenshot from 2018-05-29 17-49-07

screenshot from 2018-05-29 17-49-48

These are just a few examples project is breaking on many aspects(results not displayed for all, images and videos, etc), do we need to work on these issues after project updation or will it be fixed in this patch @hemantjadon? Considering the changes in this patch are huge.

**Comment by hemantjadon** [comment]
@simsausaurabh can you add changes to this branch and fix the changes then finally merge this PR?

The changes which I made are minimal for the transition from v4 -> v6. I can guide you to make minimal changes to work on.

**Comment by simsausaurabh** [comment]
@hemantjadon Okay, I'll update and fix the changes. @mariobehling could we create a new branch(development/project-update) to merge this patch there. After fixing all the minimal bugs created, it could be merged into master branch.

**Comment by hemantjadon** [comment]
Closing this PR here will be continued to be on development branch

**Pull request closed without merging**